server/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -214,11 +218,16 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
 
     public static boolean parseMultiField(FieldMapper.Builder builder, String name, Mapper.TypeParser.ParserContext parserContext,
                                           String propName, Object propNode) {
-        parserContext = parserContext.createMultiFieldContext(parserContext);
         if (propName.equals("fields")) {
+            if (parserContext.isWithinMultiField()) {
+                deprecationLogger.deprecatedAndMaybeLog("multifield_within_multifield", "The multi-field " +
```

Comment:
kind of a shame we are specific about a `name` for the first occurrence and then stay quiet for the rest of the offenses. Should we add a recommendation to the user to check *all* their other mappings/templates as well?

Reply:
What do you think about switching from `deprecatedAndMaybeLog(...)` to just `deprecated(...)`? This will ensure that all the deprecated usages will get logged, and there is not a big concern about logging too many messages, since this can only be triggered on a mapping update (and chained multifields should hopefully be a rare occurrence in the first place).

Reply:
> since this can only be triggered on a mapping update

are you sure? I remember testing type parsers and was surprised how many times things were parsed. this might be worth verifying.

I think `deprecatedAndMaybeLog` is appropriate. Regardless, more guidance on what to look at and how to fix the problem may be helpful.

Reply:
That's a good point, I didn't realize quite how many times this was invoked. From a quick test, a single mapping update can call this deprecation log line 4 times. I'll certainly stick with `deprecatedAndMaybeLog`.

I tried to update the deprecation message to (1) clarify that there may be other instances of chained multi-fields that the message doesn't cover, and (2) give guidance as to how to respond to the issue. Let me know if you have other suggestions!

Reply:
great! I hope that extra level of messaging is found to be helpful!